src/ingest-contact-loaders/test-fakedata/index.js

```diff
@@ -0,0 +1,112 @@
+import { r } from "../../server/models";
+import { getConfig, hasConfig } from "../../server/api/lib/config";
+
+export const name = "test-fakedata";
+
+export function displayName() {
+  return "Fake Data for Testing";
+}
+
+export function serverAdministratorInstructions() {
+  return {
+    "environmentVariables": [],
+    "description": "",
+    "setupInstructions": "Nothing is necessary to setup since this is default functionality"
+  }
+}
+
+export async function available(organization, user) {
+  /// return an object with two keys: result: true/false
+  /// if the ingest-contact-loader is usable and has
+  /// Sometimes credentials need to be setup, etc.
+  /// A second key expiresSeconds: should be how often this needs to be checked
+  /// If this is instantaneous, you can have it be 0 (i.e. always), but if it takes time
+  /// to e.g. verify credentials or test server availability,
+  /// then it's better to allow the result to be cached
+  const orgFeatures = JSON.parse(organization.features || "{}");
+  const result = (orgFeatures.service || getConfig("DEFAULT_SERVICE")) === "fakeservice"
+  return {
+    result,
+    expiresSeconds: 0
+  }
+}
+
+export function addServerEndpoints(expressApp) {
+  /// If you need to create API endpoints for server-to-server communication
+  /// this is where you would run e.g. app.post(....)
+  /// Be mindful of security and make sure there's
+  /// This is NOT where or how the client send or receive contact data
+  return;
+}
+
+export function clientChoiceDataCacheKey(organization, campaign, user) {
+  return `${organization.id}-${campaign.id}`;
+}
+
+export async function getClientChoiceData(organization, campaign, user, loaders) {
+  /// data to be sent to the admin client to present options to the component or similar
+  /// The react-component will be sent this data as a property
+  /// return a json object which will be cached for expiresSeconds long
+  /// `data` should be a single string -- it can be JSON which you can parse in the client component
+  /// `dependsOn` should be an array that is the list of arguments the result depends on
+  ///   it can include "organization", "campaign", "user" (the strings)
+  return {
+    data: `choice data from server ${Math.random()}`,
+    expiresSeconds: 0
+  };
+}
+
+export async function processContactLoad(job, maxContacts) {
+  /// trigger processing -- this will likely be the most important part
+  /// you should load contacts into the contact table with the job.campaign_id
+  /// Since this might just *begin* the processing and other work might
+  /// need to be completed asynchronously after this is completed (e.g. to distribute loads)
+  /// AFTER true contact-load completion, this (or another function) MUST call
+  /// src/workers/jobs.js::completeContactLoad(job)
+  ///   The async function completeContactLoad(job) will delete opt-outs, delete duplicate cells, clear/update caching, etc.
```

Comment:
Do we want opt-outs deleted? I thought those should live with a contact regardless of the campaign

Reply:
I should probably write it more clearly--we delete contacts that are in the opt-out list. And yes, we never want to contact optouts